linode_api4/objects/domain.py

```diff
@@ -61,3 +63,27 @@ def record_create(self, record_type, **kwargs):
 
         zr = DomainRecord(self._client, result["id"], self.id, result)
         return zr
+
+    def zone_file_view(self):
+        result = self._client.get(
+            "{}/zone-file".format(self.api_endpoint), model=self
+        )
+
+        return result["zone_file"]
+
+    def domain_clone(self, domain):
+        params = {
+            "domain": domain.domain if isinstance(domain, Domain) else domain
+        }
```

Comment:
When I tried to pass in an existing `Domain` object in the parameter, it would failed to clone.


```python
domain1 = Domain(client, 12345)
domain2 = Domain(client, 12346)

domain1.domain_clone(domain2)
```


```
linode_api4.errors.ApiError: 400: The domain another-domain.linode already exists on your account.;
```

I wonder what's the case that another Domain object can be passed in? If it is impossible, can we type the parameter to be `str` and remove the code that handle `Domain` object?

Reply:
Reading through the API docs again, I think you're right that it will be impossible to pass in a `Domain` object. I'll update the function to only accept a string. Thanks!